Var/src/var.c

```diff
@@ -0,0 +1,84 @@
+#include "var.h"
+#define PRINT_VAR(var_type, format) printf(""#format" (%p)\n", *(var_type*)ptr->valptr, ptr->valptr)
+
+#define OPTIMIZED_REALLOC(var_type, old_size, size) \
+	if (size > old_size) { \
+		/* When scaling up free/malloc is faster */ \
+		free_fn(ptr->valptr); \
+		if (!(valptr = malloc_fn(size))) { \
+			printf("Failed to allocate memory, cancelling"); \
+			return; \
+		} \
+	} \
+	else if (old_size > size){ \
+		/* When scaling down realloc is faster */ \
+		if (!(valptr = realloc(ptr->valptr, size))) { \
+			printf("Failed to reallocate memory, cancellinng\n"); \
+			return; \
+		} \
+	} \
+	else valptr = ptr->valptr; \
+
+#define ALLOC_VAL(var_type, enum_type, size)\
```

Comment:
Is the `#define` here really necessary?

Reply:
Not sure what you meant here either, to declare a macro `#define` keyword is used isnt it?

Reply:
I see that it's a macro, my question was more in the lines of "why is it a macro?" Why not a normal function? Or even an X-Macro?